lib/utils.js

```diff
@@ -146,35 +146,53 @@ function findToken (str, token) {
   return ind
 }
 
-const RDS1 = /^\.\.?\//u
-const RDS2 = /^\/\.(?:\/|$)/u
-const RDS3 = /^\/\.\.(?:\/|$)/u
-const RDS5 = /^\/?(?:.|\n)*?(?=\/|$)/u
-
-function removeDotSegments (input) {
+// RFC and Algorithm - https://datatracker.ietf.org/doc/html/rfc3986#section-5.2.4
+function removeDotSegments (path) {
+  let input = path
   const output = []
 
-  while (input.length) {
-    if (input.match(RDS1)) {
-      input = input.replace(RDS1, '')
-    } else if (input.match(RDS2)) {
-      input = input.replace(RDS2, '/')
-    } else if (input.match(RDS3)) {
-      input = input.replace(RDS3, '/')
-      output.pop()
+  while (input.length > 0) {
+    // Rule 2A: Remove leading "../" or "./"
+    if (input.startsWith('../')) {
+      input = input.slice(3)
+    } else if (input.startsWith('./')) {
+      input = input.slice(2)
+    } else if (input.startsWith('/./')) {
+      // Rule 2B: Replace "/./" or "/." with "/"
+      input = input.slice(2)
+    } else if (input === '/.') {
+      input = '/'
```

Comment:
If input = '/', then we have another iteration. We will have all checks just to do output.push(input).

So isnt it smarter to do following?

```suggestion
    } else if (input === '/.') {
      output.push('/')
      break
```